internal/e2e/instrumentation.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 SolarWinds Worldwide, LLC. All rights reserved.
```

Comment:
Minor / subjective: `infrastructure.go` sounds a bit too vague to me - maybe `containers.go` or `collector_runner.go`, ...